### PR TITLE
Added LGBN get random module and made few test changes

### LINKED ISSUE #1795: Using get_random with LinearGaussianBayesianNetwork returns incorrect model object
```python
>>> from pgmpy.models import LinearGaussianBayesianNetwork
>>> random_model = LinearGaussianBayesianNetwork.get_random(n_nodes=10, edge_prob=0.3, latents=False)
>>> random_model
<pgmpy.models.BayesianNetwork.BayesianNetwork at 0x72cd227876d0>
```

Ideally, this should return an instance of `pgmpy.models.LinearGaussianBayesianNetwork` as the get_random was called for that class.

### PR DESCRIPTION
**Pull request by kollisaisiddartha**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes #1795 

### List of changes to the codebase in this pull request
- Added get_random() function to LinearGaussianBayesianNetwork.py
- Added parameters and example for the module
- Added unit test for get_random module in test_LinearGaussianBayesianNetwork.py file

### CODE REVIEW

File: pgmpy/models/LinearGaussianBayesianNetwork.py [hunk 1]
```diff
@@ -432,3 +433,110 @@ def is_imap(self, JPD):
         raise NotImplementedError(
             "is_imap method has not been implemented for LinearGaussianBayesianNetwork."
         )
+
+    @staticmethod
+    def get_random(
+        n_nodes=5,
+        edge_prob=0.5,
+        node_names=None,
+        n_states=None,
```
Comment (ankurankan) [inline pgmpy/models/LinearGaussianBayesianNetwork.py:442]: @kollisaisiddartha `n_states` is not required for LinearGaussianBayesianNetwork. It is required by TabularCPDs which is only for BayesianNetwork.

File: pgmpy/models/LinearGaussianBayesianNetwork.py [hunk 2]
```diff
@@ -432,3 +433,110 @@ def is_imap(self, JPD):
...
+
+        edge_prob: float
+            The probability of edge between any two nodes in the topologically
+            sorted DAG.
+
+        node_names: list (default: None)
+            A list of variables names to use in the random graph.
+            If None, the node names are integer values starting from 0.
+
+        n_states: int or dict (default: None)
```
Comment (ankurankan) [inline pgmpy/models/LinearGaussianBayesianNetwork.py:465]: Same.

File: pgmpy/models/LinearGaussianBayesianNetwork.py [hunk 3]
```diff
@@ -432,3 +433,110 @@ def is_imap(self, JPD):
...
+        >>> model.edges()
+        OutEdgeView([(0, 1), (0, 2), (0, 3), (0, 4), (2, 4), (3, 4)])
+        >>> model.cpds
+        [<LinearGaussianCPD: P(0) = N(1.049; 0.882) at 0x2672d329130,
+        <LinearGaussianCPD: P(1 | 0) = N(-0.705*0 + 1.46; 0.187) at 0x26730daf9b0,
+        <LinearGaussianCPD: P(2 | 0) = N(-1.881*0 + 0.288; 1.53) at 0x267308a0800,
+        <LinearGaussianCPD: P(3 | 0) = N(1.429*0 + 0.861; 0.351) at 0x26731307620,
+        <LinearGaussianCPD: P(4 | 0, 2, 3) = N(1.103*0 + 1.812*2 + 1.397*3 + 0.877; 1.06) at 0x26730f55f10]
+        """
+        if node_names is None:
```
Comment (ankurankan) [inline pgmpy/models/LinearGaussianBayesianNetwork.py:505]: Same

File: pgmpy/models/LinearGaussianBayesianNetwork.py [hunk 4]
```diff
@@ -432,3 +433,110 @@ def is_imap(self, JPD):
...
+
+        dag = DAG.get_random(
+            n_nodes=n_nodes, edge_prob=edge_prob, node_names=node_names, latents=latents
+        )
+        lgbn_model = LinearGaussianBayesianNetwork(dag.edges(), latents=dag.latents)
+        lgbn_model.add_nodes_from(dag.nodes())
+
+        rng = np.random.default_rng(seed=seed)
+
+        cpds = []
```
Comment (ankurankan) [inline pgmpy/models/LinearGaussianBayesianNetwork.py:527]: I would rather call the `get_random_cpd` method here instead. Would reduce duplicated code.

File: pgmpy/tests/test_models/test_LinearGaussianBayesianNetwork.py [hunk 5]
```diff
@@ -195,5 +195,15 @@ def test_get_random_cpds(self):
         cpds = model_lin.get_random_cpds()
         self.assertEqual(len(cpds), len(model.nodes()))
 
+    def test_get_random(self):
+        model = LinearGaussianBayesianNetwork.get_random(
+            n_nodes=5, edge_prob=0.5, latents=False
+        )
+        self.assertIsInstance(
```
Comment (ankurankan) [inline pgmpy/tests/test_models/test_LinearGaussianBayesianNetwork.py:202]: Could you add further tests for the get_random method here? For example, testing some of the other arguments, number of nodes in the generated model, node names, etc.

**Review by ankurankan — CHANGES_REQUESTED** [review]
@kollisaisiddartha I think this works as expected now. I have added a few comments for small improvements.

File: pgmpy/models/LinearGaussianBayesianNetwork.py (hunk 3 above)
Comment (kollisaisiddartha) [reply pgmpy/models/LinearGaussianBayesianNetwork.py:505]: What should be done here?

**Comment by ankurankan** [comment]
@kollisaisiddartha BTW you don't need to close and create new PRs for changes. You can keep pushing your changes in the same branch, and it will update here. It would also help us in keeping track of our discussion.

**Comment by kollisaisiddartha** [comment]
Yeah sure will be doing that from now on

**Pull request closed without merging**